### PR TITLE
add spacing to numpy.gradient

### PR DESCRIPTION
**Pull request by epignatelli**
Hi guys, I just implemented spacing in jax.numpy.gradient.

Going to add a test shortly.

### CODE REVIEW

File: tests/lax_numpy_test.py [hunk 1]
```diff
@@ -2735,18 +2735,21 @@ def testPrecision(self):
...
+        for _num_axes in range(len(shape))
+        for varargs in [0, (1,), (2, 0)]
         for axis in itertools.combinations(range(len(shape)), _num_axes)
         for dtype in inexact_dtypes
         for rng_factory in [jtu.rand_default]))
-  def testGradient(self, shape, axis, dtype, rng_factory):
+  def testGradient(self, shape, varargs, axis, dtype, rng_factory):
     rng = rng_factory()
-    args_maker = self._GetArgsMaker(rng, [shape], [dtype])
+    args_maker = self._GetArgsMaker(rng, [shape], [dtype], [varargs])
```
Comment (mattjj) [inline tests/lax_numpy_test.py:2752]: I'm not sure this call to `_GetArgsMaker` is right: the signature is `_GetArgsMaker(self, rng, shapes, dtypes, onp_arrays=True)`, so the last argument is just meant to be a bool.

Maybe instead `varargs` should be passed in the lambdas below, like

```python
    jnp_fun = lambda y: jnp.gradient(y, varargs, axis=axis)
    onp_fun = lambda y: onp.gradient(y, varargs, axis=axis)
```

WDYT?

Comment (epignatelli) [reply tests/lax_numpy_test.py]: Ah, thanks for the clarification! 
It definitely makes sense - sorry I struggled to get the logic of the `_GetArgsMaker`, much clearer now.
I just committed that - tests are ok locally, waiting for the ci.

Comment (epignatelli) [reply tests/lax_numpy_test.py]: CI failing, it seems like the random permutation between different inputs cause `len(varargs)` to be different from `len(axis)`. What do you think? Do you forsee any other problem?

Comment (hawkinsp) [reply tests/lax_numpy_test.py]: I haven't looked into your particular problem here, but if you want `varargs` and `axis` to have equal lengths, you must change the test harness to ensure that (e.g., first generate one, and then generate the other to have the same length.)

Comment (epignatelli) [reply tests/lax_numpy_test.py]: Thanks @hawkinsp - just added that in a2fcad9. `len(varargs)` now returns `num_axes`

**Comment by epignatelli** [comment]
Hi guys, I addressed the comments, I believe this is ready for review!

**Comment by hawkinsp** [comment]
Sorry about the slow review. I know I don't personally know a lot about this particular function in Numpy. I think this is fine for a first cut, given it appears well-tested. We can do any further iteration in future PRs.

Thanks for the PR!

**Comment by epignatelli** [comment]
No worries, actually thanks guys!

I have some higher order finite difference code already working.
Is there a namespace in jax where this would be useful?
